x/auth/ante/authz_rules_spec.md

```diff
@@ -0,0 +1,65 @@
+# Description
+
+Problem: 
+The existing authorization (authz) module lacks the flexibility to grant permissions (authz grants) for various types of messages along with specific conditions or rules. This limitation constrains users from customizing their transaction behavior based on specific needs or strategies.
+
+## Specific Examples of Limitations:
+Swapping Reward Tokens:
+ - Currently, users cannot set a rule to swap their reward tokens or any other tokens for another token with a specified limit.
+Sending Tokens to Selected Addresses:
+ - Users are unable to authorize sending tokens to a pre-defined or selected address, restricting the ability to control where tokens are transferred.
+Staking Tokens with Limitations:
```

Comment:
replace it with gov usecase..

Reply:
also, consider spec format of cosmos-sdk. It describes the problem, but clearly explain about solution, conclusion etc instead of PR diff.